app/src/Backend/Bcn/BcnContext.js

```diff
@@ -0,0 +1,59 @@
+import React, { Component } from 'react';
+import BcnData from './index';
+
+const BcnContext = React.createContext();
+
+function BcnProvider({children}) {
+  return (
+    <BcnContext.Provider value={BcnData}>
+      {children}
+    </BcnContext.Provider>
+  );
+}
+
+function BcnConsumer({children}) {
+  return (
+    <BcnContext.Consumer>
+      {context => {
+        if (context === undefined) {
+          throw new Error('BcnConsumer must be used within a BcnProvider');
+        }
+
+        return children(context);
+      }}
+    </BcnContext.Consumer>
+  );
+}
+
+function useBcnData() {
+  const context = React.useContext(BcnContext);
+
+  if (context === undefined) {
+    throw new Error('useBcnData must be used within a BcnProvider');
+  }
+
+  return context;
+}
+
+function withBcnDataHandler(WrappedComponent) {
+  return (
+    class GetBcnData extends Component {
+      render() {
+        return (
+          <BcnContext.Consumer>
+            {context => {
+              if (context === undefined) {
+                throw new Error('BcnConsumer must be used within a BcnProvider');
+              }
+
+              return <WrappedComponent {...this.props} bcnDataHandler={context} />;
+            }}
+          </BcnContext.Consumer>
+        );
+      }
+    }
+  );
+}
+
+export default BcnProvider;
+export { withBcnDataHandler, BcnConsumer, useBcnData, BcnContext };
```

Comment:
This could be written smaller like:

```javascript
import React from 'react';
import BcnData from './index';

const BcnContext = React.createContext();

const BcnProvider = ({children}) => (
  <BcnContext.Provider value={BcnData}>
    {children}
  </BcnContext.Provider>
);

const useBcnData = () => {
  const context = React.useContext(BcnContext);

  if (context === undefined) {
    throw new Error('BcnData must be used within a BcnProvider');
  }

  return context;
}

const BcnConsumer = ({children}) => children( useBcnData() );

const withBcnDataHandler = (WrappedComponent) => (props) => (
  <BcnConsumer>
    { context => <WrappedComponent {...props} bcnDataHandler={context} /> }
  </BcnConsumer>
);

export default BcnProvider;
export { withBcnDataHandler, BcnConsumer, useBcnData, BcnContext };
```

Reply:
Hi @emibcn. I'll change this and the other providers as this suggestion. Regarding the discussion in the issue, I would suggest to merge after this suggestion/change and create a new issue with the improvements if that's ok with you.

Reply:
Yes, of course! Send the commits whenever you can and I'll merge it.

Reply:
Done. Since most of the changes here were related to syntax, have you thought about using ESLint to enforce arrow functions, semicolons, and format in general? 

You can also create an Issue and maybe I can tackle it down.

*Not related:* Is there a reason on putting everything inside the 'app' folder instead of having the source code in the root folder?

Reply:
**About ESLint**: I was waiting for a PR: https://github.com/emibcn/covid/pull/18
If you wanna take it, I'll open the issue with some preferred syntax (as in CONTRIBUTING.md).

**About de `app/` folder**: Yes, normally I use `docker-compose` to execute Node binaries (I could push them into the repo and add some docs...). When mounting the app code `app/` into the container as a volume (this way modifications are detected by CRA and `jest`), I'd like to NOT mount `.git`, `*.md`, etc. This way, we also have CRA README's separated from project's README.

Of course, if you have a better idea (or drawbacks I've not seen yet), please, don't hesitate to comment it.

**About this PR**: I'll manually test it now (I'm currently working on `jest` tests to stop needing this step on every PR https://github.com/emibcn/covid/pull/39 ) and merge if everything is OK.

Reply:
Ok. Everything is working. Merging!

Reply:
@sebas-alarconr : Created the ESLint issue: https://github.com/emibcn/covid/issues/41